### PR TITLE
Add possibility to get branch Model Condition

### PR DESCRIPTION
**Pull request by j0nathan33**

### CODE REVIEW

File: src/main/java/com/cdancy/bitbucket/rest/options/CreateBranchModelConfiguration.java [hunk 1]
```diff
@@ -0,0 +1,50 @@
+/*
```
Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/options/CreateBranchModelConfiguration.java]: Should this file be here? Doesn't seem to be used

File: src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java [hunk 2]
```diff
@@ -136,12 +140,49 @@ public void testDeleteBranchPermission() {
...
+    }
+
     @AfterClass
     public void fin() {
         boolean success = api().updateDefault(projectKey, repoKey, defaultBranchId);
         assertThat(success).isTrue();
         success = api().delete(projectKey, repoKey, "refs/heads/" + branchName);
         assertThat(success).isTrue();
+        if (branchModelConfiguration != null) {
+            checkDefaultBranchConfiguration();
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: Do we need to do this check twice?

File: src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java [hunk 3]
```diff
@@ -136,12 +140,49 @@ public void testDeleteBranchPermission() {
         }
     }
 
+    @Test(dependsOnMethods = {"testCreateBranch", "testListBranches"})
+    public void testGetBranchModelConfiguration() {
+        branchModelConfiguration = api().getModelConfiguration(projectKey, repoKey);
+        checkDefaultBranchConfiguration();
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: If we don't need to do this check twice than lets remove the `checkDefaultBranchConfiguration` method and instead add that code here.

File: src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java [hunk 4]
```diff
@@ -17,17 +17,16 @@
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: Another LiveTest to check for errors? Maybe named something like `testGetBranchModelConfigurationOnError()`?

File: src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java (hunk 2 above)
Comment (j0nathan33) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: Oops, This line is for update

File: src/main/java/com/cdancy/bitbucket/rest/options/CreateBranchModelConfiguration.java (hunk 1 above)
Comment (j0nathan33) [inline src/main/java/com/cdancy/bitbucket/rest/options/CreateBranchModelConfiguration.java]: I use in another PR (#69).

File: src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java (hunk 3 above)
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: Seen you pushed a commit but missed this. Not sure if you were planning on addressing with a subsequent commit.

Comment (j0nathan33) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: If I remove it, I need to readd it in another PR (#69)

Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: Gotcha ... yeah looking at that PR now. Ok lets keep it then.

File: src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java [hunk 5]
```diff
@@ -136,6 +140,48 @@ public void testDeleteBranchPermission() {
...
+        assertThat(configuration.errors()).isNotEmpty();
+        assertThat(configuration.development()).isNull();
+        assertThat(configuration.production()).isNull();
+        assertThat(configuration.types()).isEmpty();
+    }
+
+    private void checkDefaultBranchConfiguration() {
+        assertThat(branchModelConfiguration).isNotNull();
+        assertThat(branchModelConfiguration.errors().isEmpty()).isTrue();
+        assertThat(branchModelConfiguration.development().refId()).isNotNull();
```
Comment (cdancy) [inline src/test/java/com/cdancy/bitbucket/rest/features/BranchApiLiveTest.java]: This was actually null when I ran the tests. Seeing as how annotated with @Nullable not sure if we should check for it here? I'm going to merge this now and remove the check. If you have a strong objection to this let me know and we can follow up on #69

**Pull request merged**